enhancements/update/upgrade-without-registry.md

```diff
@@ -0,0 +1,893 @@
+---
+title: upgrade-without-registry
+authors:
+- "@jhernand"
+reviewers:
+- "@avishayt"
+- "@danielerez"
+- "@mrunalp"
+- "@nmagnezi"
+- "@oourfali"
+approvers:
+- "@sdodson"
+- "@zaneb"
+api-approvers:
+- "@sdodson"
+- "@zaneb"
+creation-date: 2023-06-29
+last-updated: 2023-07-26
+tracking-link:
+- https://issues.redhat.com/browse/RFE-4482
+see-also:
+- https://issues.redhat.com/browse/OCPBUGS-13219
+- https://github.com/openshift/cluster-network-operator/pull/1803
+replaces: []
+superseded-by: []
+---
+
+# Upgrade without registry
+
+## Summary
+
+Provide an automated mechanism to upgrade a cluster without requiring an image
+registry server.
+
+## Motivation
+
+All these stories are in the context of disconnected clusters with limited
+resources, both in the cluster itself and in the surrounding environment:
+
+- The cluster is not connected to the Internet, or the bandwidth is very limited.
+- It isn't possible to bring up additional machines, even temporarily.
+- The resources of the cluster are limited, in particular in terms of CPU and memory.
+- The technicians responsible for performing the upgrade have no OpenShift knowledge.
+
+These clusters are usually installed at the customer site by a partner engineer
+collaborating with customer technicians.
+
+Eventually, the cluster will need to be upgraded, and then the technicians will
+need tools that make the process as simple as possible, ideally requiring no
```

Comment:
This lack of knowledge will become a risk if something goes wrong with the upgrade. How will the customer deal with that situation?

Reply:
The technician will definitely need to know how to gather logs, but that involves running a command or two, not real OpenShift knowledge.

Reply:
The technician will have the support of the engineers that prepared the upgrade bundle. When something goes wrong it will be the responsibility of that engineer to identify and solve the issue.

Reply:
Actually a previous comment from @bparees and also other comments, made me think that it would be better to split this into two different enhancements:

1. Pre-load and pin images.
2. Upgrade bundle.

I will do that in the next pull request. I will address your suggestion in the second enhancement.

Reply:
@ingvagabund the engineer is a customer or partner engineer. He is responsible for preparing the bundle, and we assume that he has OpenShift knowledge.

I'd expect the documentation to live in the OpenShift documentation site.

When something goes wrong the customer should be asked to collect the same resources/logs that we usually collect: must gather, etc.